include/ddsrouter/types/address/Address.hpp

```diff
@@ -140,10 +168,44 @@ class Address
     //! Default transport protocol for address when is not set: UDP
     static TransportProtocol default_transport_protocol() noexcept;
 
+    /**
+     * @brief Return the IP that refers the \c domain name given with IP version specified in \c ip_version
+     *
+     * Make a DNS call to get the IP related with \c domain
+     *
+     * @param domain domain name of the ip to look for
+     * @param ip_version version of the ip to find (by default the Address ip-version default (IPv4))
```

Comment:
Maybe move the default information to the constructor below (the one without `ip_version` param)